build/build.sh

```diff
@@ -268,6 +268,8 @@ if [[ $opt == 1 ]]; then
 
   if [[ $docs == 1 ]]; then
       echo "make xrt_docs"
+      # Source the XRT environment so we can find modules like pyxrt
+      source ./opt/xilinx/xrt/setup.sh
```

Comment:
Does build.sh -docs really need to depend on having run a full build first?
Before this change, all you had to do was clone xrt; build.sh -docs,  where latter does not build XRT only docs, which makes it super fast to build/clean docs.

Reply:
sphinx-build imports the python module (pyxrt in our case) and introspects the class/methods to generate the documentation. It looks like the following:

```


class pyxrt.bo

    Represents a buffer object

    address(self: pyxrt.bo) → int

        Return the device physical address of the buffer object

    class flags

        Buffer object creation flags

        Members:

            normal

            cacheable

            device_only

            host_only

            p2p

            svm

        name

    map(self: pyxrt.bo) → memoryview

        Create a byte accessible mmemory view of the buffer object

    read(self: pyxrt.bo, arg0: int, arg1: int) → numpy.ndarray[numpy.int8]

        Read from the buffer object requested number of bytes starting from specified offset

    size(self: pyxrt.bo) → int

        Return the size of the buffer object

    sync(self: pyxrt.bo, arg0: pyxrt.xclBOSyncDirection, arg1: int, arg2: int) → None

        Synchronize (DMA or cache flush/invalidation) the buffer in the requested direction

    write(self: pyxrt.bo, arg0: buffer, arg1: int) → None

        Write the provided data into the buffer object starting at specified offset


```

Reply:
Ok, guess there is no way around this then.